hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/CustomPropertiesTest.java

```diff
@@ -65,36 +72,38 @@ public void testNativeClient() throws Exception {
         assertEquals("dev-pass", clientConfig.getGroupConfig().getPassword());
         assertTrue(clientConfig.getNetworkConfig().isSmartRouting());
         assertTrue(clientConfig.getNetworkConfig().isRedoOperation());
-        Hazelcast.newHazelcastInstance(new ClasspathXmlConfig("hazelcast-custom.xml"));
+        factory.newHazelcastInstance(config);
         assertEquals(2, hz.getCluster().getMembers().size());
         main.shutdown();
-        Thread.sleep(1000 * 3); // let client to reconnect
+        Thread.sleep(1000 * 1); // let client to reconnect
```

Comment:
Then the test will fail, I have repeated the test many times. With using `TestHazelcastFactory ` client always reconnected within 1 second.  `Thread.sleep(1000 * 3);` is too much time for mock network, 1 seconds seems enough for now.